src/utils/utils.py

```diff
@@ -52,8 +53,58 @@ def extract_children(zip_path: str, out_path: str):
 
 def create_subset_for_testing(ds: Dataset, test_set_size: int) -> Dataset:
     logger.info(f"Creating subset with size {test_set_size}")
-    num_samples = min(test_set_size, len(ds))
-    subset = Subset(ds, np.arange(num_samples))
+
+    indices_by_type: Dict[str, List[int]] = {}
+    for idx in range(len(ds)):
+        item: QueryItem = ds[idx]
+        qa_type = item.qa_type
+        indices_by_type.setdefault(qa_type, []).append(idx)
+
+    total = len(ds)
+
+    sample_count_by_type = {
+        qa_type: max(1, int(len(indices) / total * test_set_size))
+        for qa_type, indices in indices_by_type.items()
+    }
+
+    count_sum = sum(sample_count_by_type.values())
+    while count_sum < test_set_size:
```

Comment:
While this works, why not just work with percentages? That would make the code here a lot more readable, but not necessary for this PR.